File: 18-Command/src/go/light_off_command.go
package main

// LightOffCommand binds "turn off" to a Light; Undo turns it on.
type LightOffCommand struct {
	light *Light
}

func NewLightOffCommand(light *Light) *LightOffCommand {
	return &LightOffCommand{light: light}
}

func (c *LightOffCommand) Execute() { c.light.Off() }
func (c *LightOffCommand) Undo()    { c.light.On() }

File: 18-Command/src/rust/light_off_command.rs
use std::rc::Rc;

use crate::command::Command;
use crate::light::Light;

/// Concrete Command — binds "turn off" to a Light; undo turns it on.
pub struct LightOffCommand {
    light: Rc<Light>,
}

impl LightOffCommand {
    pub fn new(light: Rc<Light>) -> Self {
        LightOffCommand { light }
    }
}

impl Command for LightOffCommand {
    fn execute(&self) {
        self.light.off();
    }
    fn undo(&self) {
        self.light.on();
    }
}
